packages/webcert/src/page/SearchAndCreatePage.tsx

```diff
@@ -29,6 +30,12 @@ const CreateCertificatePage: React.FC = () => {
     ReactTooltip.hide()
   }, [patient])
 
+  useEffect(() => {
```

Comment:
Should this kind authorization logic reside in that page components? I foresee a lot of duplication for each page we create. Perhaps this kind of logic should be centralized, e.g. in a parent component or a HOC.

Reply:
Which check are you referring to, the resourceLink or the user object?

Reply:
I'm thinking of all the code in the `useEffect`, checking to see if the user is authorized to view a certain page and redirecting based on that. I've also seen some warnings relating to redirecting during rendering of a component, perhaps we should prefer to to redirects with `<Redirect />` instead of `history.push()` in components.

Reply:
For an example of what I mean, you could take a look at https://github.com/sklintyg/frontend/blob/feature/INTYGFV-14452_login_R2/packages/webcert/src/utils/withLoggedInUserRedirect.tsx